### ISSUE TITLE
Broken Travis build

### ISSUE THREAD

**Issue by UlfNorell** [agda/agda#4239]
Travis has been failing for a long time (since splitting the test suite in multiple jobs?).

Compiler tests are failing because the standard library hasn't been initialized properly (`Everything.agda` not found). Usually `make library-test` takes care of this, but now that gets run in a different job.

Latex tests are failing for reasons beyond my comprehension.

cc @L-TChen

**Comment by L-TChen** [comment]
> Usually make library-test takes care of this, but now that gets run in a different job.

Thanks, I see why it fails. I put it back to the same job. They were initially put together, but that job took much longer than others to finish. 

LaTeX tests are failing since this https://travis-ci.org/agda/agda/jobs/614750594?utm_medium=notification&utm_source=github_status

It might be caused by the new option for the code environment, @nad ?

**Comment by nad** [comment]
That test works fine on my machine. Is Travis not running the LaTeX compilers? That would explain the problem, because the golden value is `LATEX_COMPILE_FAILED with lualatex, xelatex, pdflatex`.

**Comment by L-TChen** [comment]
https://github.com/agda/agda/blob/4a84eb259281887de454dd85e0a671a64fedf30a/.travis/cabal_script#L17-L21

That might be the reason. It seems LaTeX compilers haven’t been run on Travis for ages, and the revision did not touch this part. 

Given #1675, Travis should not run LaTeX compilers either.

**Comment by L-TChen** [comment]
> Compiler tests are failing because the standard library hasn't been initialized properly (Everything.agda not found). 

This one is fixed now, see https://travis-ci.org/agda/agda/builds/617763622.

**Comment by ice1000** [comment]
I recommend disabling Travis build, because now we have github actions.

**Comment by L-TChen** [comment]
> I recommend disabling Travis build, because now we have github actions.

No. The problem with GitHub Actions is that it only allows 400MB for each cache file, but the current design is to use cache (over 600MB) to store compiled binaries AND installed libraries to run test suites in parallel. 

It is possible to restore the system without caching by uploading the artifacts to somewhere, but it requires to pack everything needed carefully.

**Comment by L-TChen** [comment]
The problem is not Travis itself, and moving all tests to GitHub does not solve the problem here.

**Comment by L-TChen** [comment]
@nad, can you revert the commit before sorting this out on another branch, or tentatively disable this specific test for the new option and leave a ticket?

**Comment by ice1000** [comment]
> The problem is not Travis itself, and moving all tests to GitHub does not solve the problem here.

But Travis is slow. We may add test suite in GitHub Actions, completely supersede Travis.
Travis have maximum 8 (or maybe 10? I can't remember the exact number) jobs, while GitHub Actions don't seem to have parallel job limit.

**Comment by L-TChen** [comment]
> But Travis is slow. We may add test suite in GitHub Actions, completely supersede Travis.
> Travis have maximum 8 (or maybe 10? I can't remember the exact number) jobs, while GitHub Actions don't seem to have parallel job limit.

The limit of concurrent jobs is 60. 

However, only 400MB for each cache file is allowed on GitHub Actions, and running Agda’s test suites requires about 600MB via caching. I am not sure if the artefacts can be made smaller, though.  

You can also work on your own fork and run Travis without problem now, since it will not exceed the time limit (50 mins). I think the limit of concurrent jobs is not a problem.

**Comment by ice1000** [comment]
Are we actually caching anything on GitHub Actions?

**Comment by L-TChen** [comment]
> Are we actually caching anything on GitHub Actions?

Not on GitHub Actions. It is used for splitting test suites in concurrent jobs on Travis.

**Comment by ice1000** [comment]
Ok. I actually meant "are we actually caching anything on Travis", but made a very misleading typo 🤣 but thanks.

**Comment by L-TChen** [comment]
@ice1000, I think it is actually possible to move the rest of testing to GitHub Actions. Since the limit is 400MB *per* cache file, we can cache dependencies and the local Agda build separately. 

It may not save us significantly more time, though. There are some overhead you need to pay for each job such as booting the virtual machine, fetching cache files, updating the package index (maybe optional), and so on. 

If you have a bit of extra time, feel free to work on this anyway. Packing a working nightly build with some libraries ready is more useful in my opinion.

**Comment by ice1000** [comment]
> If you have a bit of extra time, feel free to work on this anyway.

I wish I have. Let's see.

> Packing a working nightly build with some libraries ready is more useful in my opinion.

I didn't get -- what do you mean by "some libraries"? Is that the builtin packages, or the standard library? I've already done the builtin library packing.

**Comment by jespercockx** [comment]
> I didn't get -- what do you mean by "some libraries"? Is that the builtin packages, or the standard library? I've already done the builtin library packing.

It would be nice if we could run tests for a larger set of external libraries once per day to get a more comprehensive view on recent breakage. Some libraries that could be included:

- agda-hott (https://github.com/HoTT/HoTT-Agda)
- agda-prelude (https://github.com/UlfNorell/agda-prelude)
- some/all of the libraries by @nad (http://www.cse.chalmers.se/~nad/software.html)
- Iowa Agda Library (https://github.com/cedille/ial)
- DoCon (http://www.botik.ru/pub/local/Mechveliani/docon-A/)

Or maybe we could just do a nightly run of the latest Agda on all libraries supported by `agda-pkg` (https://github.com/agda/agda-pkg)?

**Comment by ice1000** [comment]
There's also cedille written in Agda. It's not compiling with master Agda.

**Comment by L-TChen** [comment]
> I didn't get -- what do you mean by "some libraries"? Is that the builtin packages, or the standard library? I've already done the builtin library packing.

I was thinking of the standard library, Ulf's agda-prelude, etc. as @jespercockx mentioned.

One caveat about uploading the artefact produced by GitHub Actions---these builds are currently compiled with `-O0`, so they may not be very practical to use. If you move everything to GitHub Actions, then it makes sense to store the artefact produced by this job https://travis-ci.org/agda/agda/jobs/620133530.
